rich/spinner.py

```diff
@@ -0,0 +1,47 @@
+from time import sleep
+from threading import Thread
+
+RECALL = "\033[2K\r"
+
+
+class Spinner:
+    SLASHES = "|/-\\"
+
+    def loop(self):
+        i = 0
+        while self._run:
+            print(
+                "{r}{icon} {message}".format(
+                    r=RECALL,
+                    icon=self._icons[i % len(self._icons)],
+                    message=self.message,
+                ),
+                end="",
+            )
+            i += 1
+            sleep(self._delay)
+
+    def start(self):
+        self._run = True
+        self._thread = Thread(target=self.loop)
+        self._thread.start()
+
+    def stop(self):
+        self._run = False
+        while self._thread.is_alive():
```

Comment:
Because there is nothing blocking here, this loop will use up 100% cpu while waiting for the tread to close. You can use `thread.join()` which does the same thing but won't hog the processor.